TOPページを作成する

uchihiro04 opened the pull request:
## 概要 
お部屋のTOPページを作成しました。
「はじめる」ボタンを押すとHouseViewingレコードが作成・保存され、保存されたHouseViewingレコードに紐づいた内見一覧ページに遷移します。

## スクリーンショット
* TOPページ

https://github.com/uchihiro04/house-viewing-score-log/assets/77523896/a6f9a932-eb21-45f1-97cb-e71e3064edaa

デベロッパツールで「iPhone SE」の画面（375 × 667）を想定して表示した場合

https://github.com/uchihiro04/house-viewing-score-log/assets/77523896/c6f8d60a-e65b-43d6-9054-bc713b7c2a07

## 関連Issue
- #29 

## 備考
### 実装について
以下の実装は別Issueで行います。
* デザイン

### エラーメッセージについて
スコア入力ページ・スコア編集ページでは、保存・更新に失敗した際にエラーメッセージを表示していました。

TOPページでは、ユーザーが「はじめる」ボタンを押した場合、原則HouseViewingレコードの作成・保存に失敗しないことが想定されます。そのため、TOPページへのエラーメッセージの表示は実装していません。

Close #29

uchihiro04 committed:
#29 ルーティングを設定

TOPページのルーティングを設定しました。

uchihiro04 committed:
#29 見出しと概要を表示

TOPページに見出し・概要を表示しました。アプリの説明ページとページ構成がほぼ同じのため、共通化しています。

uchihiro04 committed:
#29 使い方に関する項目を表示

uchihiro04 committed:
#29 「はじめる」ボタンの作成

TOPページに「はじめる」ボタンを作成しました。押すとHouseViewingレコードを作成・保存し、保存したHouseViewingレコードに応じた内見一覧ページに遷移します。

takuminasu commented on app/controllers/house_viewings_controller.rb:
```diff
@@ -0,0 +1,14 @@
+
+class HouseViewingsController < ApplicationController
+  def create
+    @house_viewing = HouseViewing.new
```
ローカル変数にしましょう。

takuminasu replied:
```ruby
house_viewing = HouseViewing.create
```

uchihiro04 replied:
ローカル変数でも問題なく動く場合は、インスタンス変数よりローカル変数を使ったほうがスコープを狭くできますね……。
レビューをもとに修正を行いました。該当するコミットは[4e18bd8](https://github.com/uchihiro04/house-viewing-score-log/pull/127/commits/4e18bd85d2e7d4930d07d9e745c7da6fb06257af)となります。

また夕会でご指摘いただいた通り、バリデーションの違反などによるHouseViewingレコードの新規保存の失敗は原則発生しない形になります。
そのため、saveメソッドからcreateメソッドに変更し、saveメソッドに関する条件分岐を削除しました。該当するコミットは[43ff0e0](https://github.com/uchihiro04/house-viewing-score-log/pull/127/commits/43ff0e0e60ed085033a72a366546ac409676c9b1)です。

uchihiro04 committed:
#29 インスタンス変数をローカル変数に変更

HouseViewingsコントローラのcreateアクションについて、スコープを狭めるためにインスタンス変数からローカル変数に変更しました。

uchihiro04 committed:
#29 newメソッドをcreateメソッドに変更

HouseViewingレコードの作成について、バリデーションなどによる失敗が原則発生しないため、saveメソッドをcreateメソッドに変更しました。また、saveメソッドに関する条件分岐を削除しました。

takuminasu reviewed (approved):
LGTM
